### ISSUE TITLE
user Timing

### ISSUE THREAD

**Issue by jaime-ez** [angulartics/angulartics-google-analytics#6]
As I understand there are no methods to track user timing in angulartics-ga...
Anyone working on it?
If not I could give it a try...

Saludos desde Chile :)

**Comment by seadour** [comment]
What do you mean by user timing? Is it not already a part of Google Analytics?

**Comment by jaime-ez** [comment]
I'm new to analytics so I apologize beforehand if it is already implemented.
I'm looking to use this feature in angulartics:
https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings

Looking through the code I didn't saw any method to use it...

Id like to use also:
https://developers.google.com/analytics/devguides/collection/analyticsjs/user-id

If there is already a way to use them please share how :)
Thanks!

**Comment by seadour** [comment]
Oh I see! That is a good question.

**Comment by evillemez** [comment]
@jaime-ez I haven't tested this yet, but the `userId` stuff is mentioned in the angulartics readme:

> The Google Analytics module lets you call $analytics.setUsername(username) or set up $analyticsProvider.settings.ga.userId = 'username'.

**Comment by alaz** [comment]
+1

**Comment by DanWilkerson** [comment]
tl;dr -> User Timings !== Site Speed Timings, yes this would be great to support @jaime-ez.

For the nerds:
User Timings is a separate hit type that can be used to send millisecond-scoped timings of arbitrary events into Google Analytics for visualization. In the reporting interface, User Timings are found under Behavior -> Site Speed -> User TImings.

They're different than the Page Speed timings; those are enabled by default, as @seadour mentions, and those simply poll the Navigation Timing APIs and send that data to GA. By default, 1% of users also send page speed timings when they generate pageview hits, but you can change this by setting siteSpeedSampleRate on your Tracker, which I don't believe is currently supported in angulartics. The docs for the old version of GA (Classic) stipulate a maximum of 10k hits or 1% of users tracked for Site Speed hits, whichever is smaller. Universal Analytics doesn't have this documented explicitly anywhere I've seen/heard, so I would assume the same rules apply here. Note that all of this happens on the server-side of GA, so if you set siteSpeedSampleRate to 100, you'll *see* timings go out with every pageview, but GA will only process a percentage of them.

User Timings will be processed variable amounts of time, depending on the [volume of pageviews your property collected the previous day](https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings#sampling_considerations). Generally speaking, your limit will be 10% of pageview hits or 10,000 hits, whichever you reach first. You can sample these by using siteSpeedSampleRate or sampleRate. I strongly recommend against using sampleRate, as if set incorrectly, you can sample ALL data you collect instead of just an individual timing.

**Comment by timelf123** [comment]
https://github.com/angulartics/angulartics/pull/413 adds the userTiming handler base needed to pull this off.

As always, we welcome anybody to submit a PR that finishes this up
